frontend/src/app/components/NewsUpdatesDialog/index.js

```diff
@@ -63,102 +67,63 @@ export default class NewsUpdatesDialog extends React.Component {
   }
 
   render() {
-    const { newsUpdates } = this.props;
+    const { sendToGA, newsUpdates } = this.props;
     const { currentStep } = this.state;
 
-    const atStart = (currentStep === 0);
-    const atEnd   = (currentStep === newsUpdates.length - 1);
-    const update = this.renderUpdate(newsUpdates, currentStep);
-    const headerTitle = this.renderHeaderTitle(newsUpdates, currentStep);
-
-    return (
-      <div className='modal-container'>
-        <div className='modal news-updates-modal'>
-          <header className='modal-header-wrapper'>{headerTitle}</header>
-          {update}
-          <div className='tour-actions'>
-            <div onClick={() => this.stepBack()} className={cn('tour-back', { hidden: atStart })}><div/></div>
-            <div onClick={() => this.stepNext()} className={cn('tour-next', { 'no-display': atEnd })}><div/></div>
-            <Localized id='tourDoneButton'>
-              <div onClick={e => this.complete(e)} className={cn('tour-done', { 'no-display': !atEnd })}>Done</div>
-            </Localized>
-          </div>
-          <div className="tour-general-actions">
-            <div className='modal-skip' onClick={e => this.cancel(e)}>Skip</div>
-            <div className="dot-wrap"><div className="dot-row">{this.renderDots(newsUpdates, currentStep)}</div></div>
-          </div>
-        </div>
-      </div>
-    );
+    const myProps = {
```

Comment:
Not a huge fan of this construct. It will get rebuilt from scratch whenever the component re-renders - which can happen quite often. Maybe isn't a practical issue here, but is generally not great. 

All of these functions bound to this can be changed over to [class arrow functions](https://medium.freecodecamp.org/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56) like [over here in the Experiment Page container](https://github.com/mozilla/testpilot/blob/master/frontend/src/app/containers/ExperimentPage/DetailsHeader.js#L243)

Reply:
@lmorchard Does that mean the "this" they are bound to is the class, not the instance? The instances don't yet exist at class declaration time.

Reply:
Bound to the instance.